pkg/metrics/metrics.go

```diff
@@ -87,64 +84,12 @@ func SetRHMIInfo(installation *integreatlyv1alpha1.RHMI) {
 	)
 }
 
+// SetRHMIStatus exposes rhmi_status metric for each stage
 func SetRHMIStatus(installation *integreatlyv1alpha1.RHMI) {
-
-	isCompleted := float64(0)
-	if installation.Status.Stage == "complete" {
-		isCompleted = float64(1)
-	}
-
-	// creates the metric labels with values
-	labelsWithValue := make(map[string]string, len(rhmiStatusLabels))
-	for _, label := range rhmiStatusLabels {
-		labelsWithValue[label] = ""
-	}
-
-	// sets value from rhmi installation to labels
-	labelsWithValue["operator_name"] = installation.GetName()
-	labelsWithValue["namespace"] = installation.GetNamespace()
-	labelsWithValue["last_error"] = installation.Status.LastError
-	labelsWithValue["preflight_message"] = installation.Status.PreflightMessage
-	labelsWithValue["preflight_status"] = string(installation.Status.PreflightStatus)
-	labelsWithValue["stage"] = string(installation.Status.Stage)
-
-	for _, stage := range installation.Status.Stages {
-		for _, product := range stage.Products {
-			labelsWithValue[SanitizeForPrometheusLabel(product.Name)] = string(product.Status)
-		}
-	}
-
-	if RHMIStatus != nil {
-		RHMIStatus.Reset()
-		RHMIStatus.With(labelsWithValue).Set(isCompleted)
-	}
-}
-
-func ExposeRHMIStatusMetric(stages []integreatlyv1alpha1.RHMIStageStatus) {
-
-	if RHMIStatus == nil {
-		for _, stage := range stages {
-			for _, product := range stage.Products {
-				rhmiStatusLabels = append(rhmiStatusLabels, SanitizeForPrometheusLabel(product.Name))
-			}
-		}
-
-		RHMIStatus = prometheus.NewGaugeVec(
-			prometheus.GaugeOpts{
-				Name: "rhmi_status",
-				Help: "RHMI status for when installation completes",
-			},
-			rhmiStatusLabels,
-		)
-		customMetrics.Registry.MustRegister(RHMIStatus)
-	}
-}
-
-func SanitizeForPrometheusLabel(productName integreatlyv1alpha1.ProductName) string {
-	if productName == integreatlyv1alpha1.Product3Scale {
-		productName = "threescale"
+	RHMIStatus.Reset()
+	if string(installation.Status.Stage) != "" {
+		RHMIStatus.With(prometheus.Labels{"stage": string(installation.Status.Stage)}).Set(float64(1))
```

Comment:
If you have empty stage, you probably still want to report the metric, right?  Is there meaning to "installed and this code is running but no stage is set" such that not reporting the value makes sense?

Reply:
I might suggest avoiding reset and just setting the value always to `"stage": string(installation.Status.Stage)`.  That feels more natural use of the "status" label.  If stage == "" has special meaning, and you want the value of the metric to be zero, then it makes the code more explicit with an else block (instead of reset).

Reply:
@maleck13 or @philbrookes What are your thoughts on this?
What does an empty stage mean?

> I might suggest avoiding reset and just setting the value always to `"stage": string(installation.Status.Stage)`

@smarterclayton I might be using the prom library incorrectly, but if the metric isn't reset it results in the number of series going up by 1 each time the stage progresses. It's accumulative.

e.g.
scrape after startup
```
rhmi_status{stage=""} 1
```
scrape after <1m running
```
rhmi_status{stage=""} 1
rhmi_status{stage="Preflight Checks"} 1
```
scrape after >2m running
```
rhmi_status{stage=""} 1
rhmi_status{stage="Preflight Checks"} 1
rhmi_status{stage="cloud-resources"} 1
```
and so on until complete & there are 8 series.

The series seem to be held in memory.
So if the operator is killed/restarted, the metric is emptied of all series and starts building them up from whatever the current stage is.
This seems like odd/undesired behaviour vs. only have 1 series exposed at all times with the current stage, or have 0's for series unless it's the current stage.

Reply:
stage being empty is what would happen before bootstrap has started. This stage shouldn't last long, but could be prolonged by failing preflight checks. I'd be reasonably happy with the statement `"" == "preflight"`in this sense. 

@Boomatang you initially implemented the stage field in the CR, do you agree?

Reply:
The stage field in the CR is set a the start of each reconcile stage if that stage is in progress. When starting there is only a very short time that the stage has no value till the preflight checks start. If for some reason the preflight checks fail the CR stage value will stay as `"Prefight Checks"`. I don't see any way once the CR stage is set that it can go back to been a blank value.

The one caveat to be aware of is the stage value is set in the CR if the reconciler reports a stage is in progress. For example RHMI is full installed and the CR stage is marked as "completed". During the reconcile loop the "cloud-resources" actions start. Firstly the stage checks if the reconciler reports a "in progress" status for the "cloud-resources" and updates the CR if required. If during the rest of the "cloud-resources" actions the status gets set to "in progress" the CR stage will not be updated till the next pass of the reconcile loop. This is also true when doing an install. The CR stage may report the previous stage while starting to install the next stage but will be updated with the correct values on the next pass of the reconcile loop.

In short I agree with @philbrookes the length of time the CR stage value can be empty is so short there is no need to do anything with this. And sense this is a metric been reported my guess is we would have gotten pass the "preflight" stage I don't see how the CR stage value can ever be set back to a blank value.